### ISSUE TITLE
accessibility issue with screen reader

### ISSUE THREAD

**Issue by sine-w** [mirinsoft/debotnet#25]
Hi,
This is a screen reader user.
I use NVDA to navigate the program and found that I cannot perform the basic function.

specifically, nvda can found a list view listing all the functions, use up and down arrow can navigate to different functions, e.g. Turn off Windows updates sharing
However tab to different part of the window seemed to be problematic, for example one some parts nvda reported as unknown;
one part nvda report "all" but nothing else;
the only part that is accessible to nvda is the part displaying the description of the function, e.g. Turn off Windows updates sharing

And I found no way to activate the function with nvda screen reader.

Please try to address this issue, thanks.

**Comment by Belim** [comment]
Hi,

unfortunately I have no experience with this software.  Debotnet is designed as Desktop software and for mouse operations. I've never tested it with a screenreader software. I'll put this on my to-do list but unfortunately cannot help here at the moment.

What about comparable programs? Can these always be controlled without problems? I wouldn't even know which interfaces have to be operated. I think it should be Microsoft Active Accessibility/Microsoft UI Automation Interface.

Regards from Germany
Belim

**Comment by lstar4ever** [comment]
> unfortunately I have no experience with this software. Debotnet is designed as Desktop software and for mouse operations. I've never tested it with a screenreader software. I'll put this on my to-do list but unfortunately cannot help here at the moment.

I understood.



> What about comparable programs? Can these always be controlled without problems

I am sorry, your software is the only one that I have tried out. I fear those close sourced Win 10 configurating script.

>I wouldn't even know which interfaces have to be operated. I think it should be Microsoft Active Accessibility/Microsoft UI Automation Interface.

Cause I am not a programmer expert, I cannot answer your question.
I have bring this issue to our mailing list, and hope someone with programming knowledge may answer here.

**Comment by blindman75** [comment]
you are correct, in that you will need to tap in to UI Automation.
The API is well documented, so working with it shouldn't be an issue.
It's not just that though, you would need to insure that buttons, drop down list boxes, text edit boxes, tick boxes and various other controls are clearly labeled and accessible to whichever screenreader is installed on the machine.

We can use a mouse if we really have to, but, on account of being blind, mouse use it not as easy for us as it requires a level of hand-eye coordination and fine motor control that for obvious reasons, we severely lack. I am happy to assist where I can, however, my programming skills are non-existent, thus, I'm happy to test as an end-user and provide you with constructive feedback.

**Comment by JaffarSidekJaffar** [comment]
Hi.  Yes those two comments are right on the money.  NVDA is a screen reader built on Python 3, with wxPython as the GUI base.  It relies heavily on Microsoft's UIA accessibility model as it's accessibility architecture, as with most native windows apps with accessibility already built in.  Is your program open source?  If it is, I could take a look at it and see how NVDA could be made to access your program as I am a programmer myself.  From one of the comments, i see that your program could be very helpful in fulfilling some of the functions not readily accessible, so i would like to help in any way I can.  Thank you and cheers.

**Comment by Belim** [comment]
I will not provide an fix for here as the development of Debotnet will be abandoned this year
http://www.mirinsoft.com/blog/19-apps/27-going-forward-with-sharpapp-only
